src/mixins/login-mixin.ts

```diff
@@ -89,6 +89,10 @@ const loginMixin = <MixinBase extends typeof PuppetSkeleton>(mixinBase: MixinBas
           contactId: this.currentUserId,
           data: 'puppet stop()',
         })
+        await new Promise<void>(resolve => setImmediate(() => {
```

Comment:
Can we just do `setImmediate(() => this.__currentUserId = undefined)`, or `this.__currentUserId = undefined` to make the code simpler?

Reply:
I did this becuase in ```logout()``` you said 

> /**
       * Huan(202111): We postpone the `this._currentUserId = undefined` to here,
       *  in case of the `logout` event listener need to check the `this.currentUserId`
       */